tests/integration/test_writes/test_rewrite_manifests.py

```diff
@@ -0,0 +1,249 @@
+
+    tbl.refresh()
+
+    current_snapshot = tbl.current_snapshot()
+    if not current_snapshot:
+        raise AssertionError
+    new_manifests = current_snapshot.manifests(tbl.io)
+    assert len(new_manifests) == 1, "Should have 1 manifest after rewrite"
+    assert new_manifests[0].existing_files_count == 2, "Should have 4 files in the new manifest"
+    assert new_manifests[0].added_files_count == 0, "Should have no added files in the new manifest"
+    assert new_manifests[0].deleted_files_count == 0, "Should have no deleted files in the new manifest"
+
+    # Validate the records
+    expected_records_count = arrow_table_with_null.shape[0] * 2
+    result_df = tbl.scan().to_pandas()
+    actual_records_count = result_df.shape[0]
+    assert expected_records_count == actual_records_count, "Rows must match"
+
+
+def compute_manifest_entry_size_bytes(manifests: List[ManifestFile]) -> float:
+    total_size = 0
+    num_entries = 0
+
+    for manifest in manifests:
+        total_size += manifest.manifest_length
+        num_entries += manifest.added_files_count + manifest.existing_files_count + manifest.deleted_files_count
+
+    return total_size / num_entries if num_entries > 0 else 0
+
+
+def test_rewrite_small_manifests_partitioned_table(session_catalog: Catalog) -> None:
+    records1 = pa.Table.from_pydict({"c1": [1, 1], "c2": [None, "BBBBBBBBBB"], "c3": ["AAAA", "BBBB"]})
+
+    records2 = records2 = pa.Table.from_pydict({"c1": [2, 2], "c2": ["CCCCCCCCCC", "DDDDDDDDDD"], "c3": ["CCCC", "DDDD"]})
+
+    records3 = records3 = pa.Table.from_pydict({"c1": [3, 3], "c2": ["EEEEEEEEEE", "FFFFFFFFFF"], "c3": ["EEEE", "FFFF"]})
+
+    records4 = records4 = pa.Table.from_pydict({"c1": [4, 4], "c2": ["GGGGGGGGGG", "HHHHHHHHHG"], "c3": ["GGGG", "HHHH"]})
+
+    schema = pa.schema(
+        [
+            ("c1", pa.int64()),
+            ("c2", pa.string()),
+            ("c3", pa.string()),
+        ]
+    )
+
+    identifier = "default.test_rewrite_small_manifests_non_partitioned_table"
+    tbl = _create_table(session_catalog, identifier, {"format-version": "2"}, schema=schema)
+
+    tbl.append(records1)
+    tbl.append(records2)
+    tbl.append(records3)
+    tbl.append(records4)
+    tbl.refresh()
+
+    tbl.refresh()
+    manifests = tbl.current_snapshot().manifests(tbl.io)
+    assert len(manifests) == 4, "Should have 4 manifests before rewrite"
+
+    # manifest_entry_size_bytes = compute_manifest_entry_size_bytes(manifests)
+    target_manifest_size_bytes = 5200 * 2 + 100
+    tbl = (
+        tbl.transaction()
+        .set_properties({TableProperties.MANIFEST_TARGET_SIZE_BYTES: str(target_manifest_size_bytes)})
+        .commit_transaction()
+    )
+
+    result = tbl.rewrite_manifests()
+
+    tbl.refresh()
+    assert len(result.rewritten_manifests) == 4, "Action should rewrite 4 manifests"
+    assert len(result.added_manifests) == 2, "Action should add 2 manifests"
+
+    new_manifests = tbl.current_snapshot().manifests(tbl.io)
```

Comment:
Nothing wrong with this, but `tbl.inspect` might also be helpful to assert the manifests.